xdem/dem.py

```diff
@@ -0,0 +1,163 @@
+"""
+dem.py provides a class for working with digital elevation models (DEMs)
+"""
+import pyproj
+import warnings
+from geoutils.satimg import SatelliteImage
+from pyproj import Transformer
+
+def parse_vref_from_product(product):
+
+    # sources for defining vertical references:
+    # AW3D30: https://www.eorc.jaxa.jp/ALOS/en/aw3d30/aw3d30v11_format_e.pdf
+    # SRTMGL1: https://lpdaac.usgs.gov/documents/179/SRTM_User_Guide_V3.pdf
+    # SRTMv4.1: http://www.cgiar-csi.org/data/srtm-90m-digital-elevation-database-v4-1
+    # ASTGTM2/ASTGTM3: https://lpdaac.usgs.gov/documents/434/ASTGTM_User_Guide_V3.pdf
+    # NASADEM: https://lpdaac.usgs.gov/documents/592/NASADEM_User_Guide_V1.pdf !! HGTS is ellipsoid, HGT is EGM96 geoid !!
+    # ArcticDEM (mosaic and strips): https://www.pgc.umn.edu/data/arcticdem/
+    # REMA (mosaic and strips): https://www.pgc.umn.edu/data/rema/
+    # TanDEM-X 90m global: https://geoservice.dlr.de/web/dataguide/tdm90/
+    # COPERNICUS DEM: https://spacedata.copernicus.eu/web/cscda/dataset-details?articleId=394198
+
+    if product in ['ArcticDEM/REMA','TDM1','NASADEM-HGTS']:
+        vref = 'WGS84'
+    elif product in ['AW3D30','SRTMv4.1','SRTMGL1','ASTGTM2','NASADEM-HGT']:
+        vref = 'EGM96'
+    elif product in ['COPDEM']:
+        vref = 'EGM08'
+    else:
+        vref = None
+
+    return vref
+
+class DEM(SatelliteImage):
+
+    def __init__(self, filename, vref=None, vref_grid=None, ccrs = None, read_vref_from_prod=True, silent=False, **kwargs):
+
+        super().__init__(filename, **kwargs)
+
+        if self.nbands > 1:
+            raise ValueError('DEM rasters should be composed of only one band only')
+
+        # priority to user input
+        self.vref = vref
+        self.vref_grid = vref_grid
+        self.ccrs = ccrs
+
+        # trying to get vref from product name
+        if read_vref_from_prod:
+            self.__parse_vref_from_fn(silent=silent)
+
+
+    def __parse_vref_from_fn(self,silent=False):
+
+        """
+        Attempts to pull vertical reference from product name identified by SatImg
+        """
+
+        if self.product is not None:
```

Comment:
self.product is never set anywhere.

Reply:
it is set in the SatImg class, so through:
`        super().__init__(filename, attrs=attrs, load_data=load_data, bands=bands, as_memfile=as_memfile)
`